### PR TITLE
Refactor(output-json): Fixed some json objects with errors in PR-#3228.

### LINKED ISSUE #3313: There are some error objects in the output json, and the json format needs to be adjusted.
### Describe the Code Quality Issue

The output of the abacus json object designed in issue3014 has some incorrect values, and the json format is not indented, which is not conducive to viewing.

### PR DESCRIPTION
**Pull request by grysgreat**
Added document support for the rapidjson library.

### Linked Issue
Fix #3313

### What's changed?
- docs : add rapidjson support in easy install document
- c++files : fix json object bugs
### Any changes of core modules? (ignore if not applicable)
- None

### CODE REVIEW

File: docs/quick_start/easy_install.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ These requirements support the calculation of plane-wave basis in ABACUS. For LC
 Some of these packages can be installed with popular package management system, such as `apt` and `yum`:
 
 ```bash
-sudo apt update && sudo apt install -y libopenblas-openmp-dev liblapack-dev libscalapack-mpi-dev libelpa-dev libfftw3-dev libcereal-dev libxc-dev g++ make cmake bc git
+sudo apt update && sudo apt install -y libopenblas-openmp-dev liblapack-dev libscalapack-mpi-dev libelpa-dev libfftw3-dev libcereal-dev libxc-dev g++ make cmake bc git rapidjson-dev
```
Comment (caic99) [inline docs/quick_start/easy_install.md:31]: Please address requirement explicitly.

**Review by caic99 — APPROVED** [review]

File: docs/quick_start/easy_install.md (hunk 1 above)
Comment (caic99) [inline docs/quick_start/easy_install.md:31]: Please address requirement explicitly.

**Review by caic99 — CHANGES_REQUESTED** [review]

File: docs/quick_start/easy_install.md (hunk 1 above)
Comment (grysgreat) [reply docs/quick_start/easy_install.md:31]: This is the support rapidjson library needed to add new json output and can be installed directly from the command line. Do I need to add comments in doc?

Comment (caic99) [reply docs/quick_start/easy_install.md:31]: > This is the support rapidjson library needed to add new json output and can be installed directly from the command line. Do I need to add comments in doc?

Yes. Installing by `apt` is not the only method for the users. In fact most supercomputer users can not use `apt`.

**Pull request closed without merging**